src/coreclr/tools/Common/JitInterface/JitConfigProvider.cs

```diff
@@ -142,10 +141,19 @@ private static string GetTargetSpec(TargetDetails target)
                 _ => throw new NotImplementedException(target.Architecture.ToString())
             };
 
-            if ((target.Architecture == TargetArchitecture.ARM64) || (target.Architecture == TargetArchitecture.ARM))
+            string targetOSComponent;
+            if (target.Architecture is TargetArchitecture.ARM64 or TargetArchitecture.ARM)
             {
                 targetOSComponent = "universal";
             }
+            else if (target.OperatingSystem == TargetOS.Windows && target.Architecture == TargetArchitecture.X86 && target.Abi == TargetAbi.NativeAot)
```

Comment:
Hmm. R2R sets the ABI to `Crossgen2RootCommand.IsArmel ? TargetAbi.NativeAotArmel : TargetAbi.NativeAot`. I did not expect that.

Reply:
@MichalStrehovsky is this intentional?

As far as I can see the expected value would be `TargetAbi.Jit` (ie. match what `CorInfoImpl` reports to RyuJIT). There are couple of checks for unsupported ARMel platform and this one branch, which apparently doesn't take place today:
https://github.com/dotnet/runtime/blob/b29efed9995da044d97f9d3f97bc3d9de80a2050/src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs#L954-L955

Reply:
TargetAbi.Jit is for the native AOT JIT prototype. Dead code. The ABI=armel is also weird but sigh.

Ref: https://github.com/dotnet/runtime/pull/58552#discussion_r701584141
Ref: https://github.com/dotnet/runtime/pull/43706#pullrequestreview-515494769

You can use the `#if READYTORUN` ifdef to decide between crossgen2 and ilc in this file.